dFusion/dFusionSpec.md

```diff
@@ -50,43 +48,48 @@ function appendOrders( bytes32 [] orders){
 
 We will use a snark to do this job:
 ```
 Snark - TransitionHashes&Validation ( public input: orderHashSha,
 					Private input: [orders])
-					Output: orderHashPederson
+					Output: orderHashPedersen
 ```
 The transitionHashes&Validation snark will do the following checks:
-- Verify the private input by recalculating the sha of all orders and comparing it to the public input `orderHashSha`.
-- Iterate over all orders again and hash them sequencially using the Pederson hash. Use this hash as output.
-
-Notice that we allow orders, which might not be covered by any balance of the order sender. These orders will be sorted out later in the settlement of an auction.
+- Verify the private input by recalculating SHA256 of all orders and comparing it to the public input `orderHashSha`.
+- Iterate over all orders again and hash them sequencially using the Pedersen hash. Use this hash as output.
 
-In the anchor contract, we have the following functionality for this process:
+Since computing the actual snark proof is very time-intense we optimistically accept state transitions that provide a significant bond instead of the actual proof.
 
-Anyone can propose a transition to the anchor contract by providing the required information and by providing a very significant bond. It is not required to provide the snark in the first place:
+Anyone can propose a transition to the anchor contract by providing the required information:
 
 ```js
-Function submitTransitionInformation( bytes32 oldstate, bytes32 newstate)
+Function submitTransitionInformation(bytes32 orderHashPedersen)
 ```
 
-In case the send-transition information is incorrect, anyone can challenge it by also providing a significant bond and calling the following function.
+In case the information is incorrect, anyone can challenge it by also providing a significant bond and calling the following function.
+
 ```js
-Function challengeTransitionInformation( bytes32 oldstate, bytes32 newstate)
+Function challengeTransitionInformation(bytes32 orderHashPedersen)
 ```
-Any significantly bonded challenge is, by default, assumed to be legitimate and will be executed after a certain time frame (some ours), unless the first transition submitter can provide a snark proof of correctness within this predefined time frame.
+Any significantly bonded challenge is, by default, assumed to be legitimate and will be executed after a certain time frame (some hours), unless the first transition submitter can provide a snark proof of correctness within this predefined time frame.
 
-The snark would be evaluated by the anchor contract after calling the following function.
+The snark will be evaluated by the anchor contract after calling the following function. The contract will populate public inputs and outputs to the snark with the data from the challenged submission.
  ```js
-Function submitSnarkToResolveChallenge(bytes32 oldstate, bytes32 newstate, --snark--)
+Function submitSnarkToResolveChallenge(--snark--)
```

Comment:
We need to refer to some state transition, in case there are several challenges ongoing

Reply:
Will also use batchId